src/encoder.ts

```diff
@@ -26,16 +26,15 @@ export function dateToTimestampProto(timeString) {
   }
   let date = new Date(timeString);
   let seconds = Math.floor(date.getTime() / 1000);
-  let nanoString = timeString.substring(20, timeString.length - 1);
-
-  if (nanoString.length === 3) {
-    nanoString = `${nanoString}000000`;
-  } else if (nanoString.length === 6)  {
-    nanoString = `${nanoString}000`;
+  let nanos = 0;
+  if (timeString.length > 20) {
+    const nanoString = timeString.substring(20, timeString.length - 1);
+    const trailingZeroes = 9 - nanoString.length;
+    nanos = parseInt(nanoString, 10) * Math.pow(10, trailingZeroes);
   }
-  let proto = {
+
+  return {
```

Comment:
You can also say `return {seconds, nanos}`
It felt weird to me at first, but it starts to feel pretty natural and expressive. It also encourages you to use variable names as they'll appear in your return values/API calls which tends to lead to more readable code.

Reply:
nifty! thanks for the tip!